webviz_ert/data_loader/__init__.py

```diff
@@ -191,32 +191,62 @@ def get_ensemble_userdata(self, ensemble_id: str) -> dict:
     def get_ensemble_parameters(self, ensemble_id: str) -> list:
         return self._get(url=f"ensembles/{ensemble_id}/parameters").json()
 
+    def get_record_labels(self, ensemble_id: str, name: str) -> list:
+        return self._get(url=f"ensembles/{ensemble_id}/records/{name}/labels").json()
+
     def get_experiment_priors(self, experiment_id: str) -> dict:
         return json.loads(
             self._query(GET_PRIORS, id=experiment_id)["experiment"]["priors"]
         )
 
     def get_ensemble_parameter_data(
-        self, ensemble_id: str, parameter_name: str
+        self,
+        ensemble_id: str,
+        parameter_name: str,
+        realizations: list,
     ) -> pd.DataFrame:
-        resp = self._get(
-            url=f"ensembles/{ensemble_id}/records/{parameter_name}",
-            headers={"accept": "application/x-parquet"},
-        )
-        stream = io.BytesIO(resp.content)
-        df = pd.read_parquet(stream)
-        return df
+        try:
+            if "::" in parameter_name:
+                name, label = parameter_name.split("::", 1)
+                resp = self._get(
+                    url=f"ensembles/{ensemble_id}/records/{name}",
+                    headers={"accept": "application/x-parquet"},
+                    params={"label": label},
+                )
+            else:
+                resp = self._get(
+                    url=f"ensembles/{ensemble_id}/records/{parameter_name}",
+                    headers={"accept": "application/x-parquet"},
+                )
+            stream = io.BytesIO(resp.content)
+            df = pd.read_parquet(stream)
+            return df
+        except DataLoaderException as e:
+            logger.error(e)
+            return self.get_ensemble_record_data(
```

Comment:
Is this necessary because the above only fetches ensemble wide data, and this call fetches realization by realization?

Reply:
Yes, that's right